client/allocrunnerv2/taskrunner/logmon_hook.go

```diff
@@ -0,0 +1,101 @@
+package taskrunner
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"runtime"
+
+	hclog "github.com/hashicorp/go-hclog"
+	plugin "github.com/hashicorp/go-plugin"
+	"github.com/hashicorp/nomad/client/allocdir"
+	"github.com/hashicorp/nomad/client/allocrunnerv2/interfaces"
+	"github.com/hashicorp/nomad/client/logmon"
+	"github.com/hashicorp/nomad/helper/uuid"
+)
+
+// logmonHook launches logmon and manages task logging
+type logmonHook struct {
+	// logmon is the handle to the log monitor process for the task.
+	logmon             logmon.LogMon
+	logmonPluginClient *plugin.Client
+
+	stdoutFifo string
+	stderrFifo string
+
+	taskDir *allocdir.TaskDir
+
+	logger hclog.Logger
+}
+
+func newLogMonHook(tr *TaskRunner, logger hclog.Logger) *logmonHook {
+	hook := &logmonHook{
+		taskDir: tr.taskDir,
+		logger:  logger,
+	}
+
+	tr.taskLoggingFifoGetter = hook.getFifos
```

Comment:
I think we should invert this relationship to avoid `taskLoggingFifoGetter` ever being a nil func. Put a setter on TaskRunner to receive the FIFOs: https://github.com/hashicorp/nomad/blob/22fa055a6e810e883606b33a1dd2028ded1f8df8/client/allocrunnerv2/taskrunner/task_runner.go#L689-L697

You can ever pass it via an interface to make hook testing easy:
https://github.com/hashicorp/nomad/blob/22fa055a6e810e883606b33a1dd2028ded1f8df8/client/allocrunnerv2/taskrunner/stats_hook_test.go#L32-L36

Reply:
(I'm not tied to this approach. "Polluting" our runners with a bunch of callbacks has its own downsides, so I'm open to ideas.)